ui/store/actions.ts

```diff
@@ -2825,7 +2823,9 @@ export function displayWarning(payload: unknown): PayloadAction<string> {
   if (isErrorWithMessage(payload)) {
     return {
       type: actionConstants.DISPLAY_WARNING,
-      payload: payload.message,
+      payload:
+        (payload.data as OptionalDataWithOptionalCause)?.cause?.message ||
```

Comment:
Nit: We should be able to avoid a cast here as well, by using `hasProperty` and `isObject` from `@metamask/utils`